include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -56,17 +60,19 @@ inline NSGA2::NSGA2(const size_t populationSize,
     mutationProb(mutationProb),
     mutationStrength(mutationStrength),
     epsilon(epsilon),
-    lowerBound(lowerBound * arma::ones(1, 1)),
-    upperBound(upperBound * arma::ones(1, 1))
+    lowerBound(lowerBound * ColType(1, GetFillType<MatType>::ones)),
+    upperBound(upperBound * ColType(1, GetFillType<MatType>::ones))
```

Comment:
Couldn't you also do `lowerBound({ lowerBound })`?  I think that should work for both Armadillo and Bandicoot.